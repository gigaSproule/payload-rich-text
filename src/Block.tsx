import type { Options } from "./types";

export type PayloadBlockFields = {
  blockType: string;
  [key: string]: unknown;
};

export type GraphQlPayloadBlockFields = {
  blockType?: string;
  [key: string]: unknown;
};

export type BlockData<T extends PayloadBlockFields = PayloadBlockFields> = {
  type: "block";
  format: string;
  version: number;
  fields: T;
};

export type GraphQlBlockData<
  T extends GraphQlPayloadBlockFields = GraphQlPayloadBlockFields,
> = {
  type: "block";
  format: string;
  version?: number;
  fields: T;
};

export type Props = {
  data: BlockData | GraphQlBlockData;
  options?: Options;
};

export const Block = ({ data, options }: Props) => {
  return options?.block ? options.block(data) : undefined;
};
