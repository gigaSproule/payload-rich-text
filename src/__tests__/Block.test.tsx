import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../Block";
import { Block } from "../Block";
import { createBlockData } from "./helpers/createBlockData";

describe("Block", () => {
  it("should not render anything if no options provided", () => {
    const blockData: Props["data"] = createBlockData();
    const { container } = render(<Block data={blockData} />);
    expect(container).toMatchSnapshot();
  });

  it("should not render anything if options provided without block", () => {
    const blockData: Props["data"] = createBlockData();
    const { container } = render(<Block data={blockData} options={{}} />);
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with block", () => {
    const blockData: Props["data"] = createBlockData();
    const { container } = render(
      <Block
        data={blockData}
        options={{
          block: (block) => {
            return <div>Custom Block - {JSON.stringify(block)}</div>;
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
