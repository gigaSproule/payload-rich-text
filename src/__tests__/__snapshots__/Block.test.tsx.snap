// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Block should not render anything if no options provided 1`] = `<div />`;

exports[`Block should not render anything if options provided without block 1`] = `<div />`;

exports[`Block should render custom render if options provided with block 1`] = `
<div>
  <div>
    Custom Block - 
    {"format":"","type":"block","version":2,"fields":{"blockType":"some-block-type","some":"custom","block":"fields"}}
  </div>
</div>
`;
